Upgrading Spark->3.3,Hadoop->3,Scala->2.13,Java->17 (#1727)
jupyter/docker-stacks: docs/using/specifics.md

You can build a `pyspark-notebook` image (and also the downstream `all-spark-notebook` image) with a different version of Spark by overriding the default value of the following arguments at build time.

> - Spark distribution is defined by the combination of Spark, Hadoop and Scala version and verified by the package checksum,

Reviewer: ```suggestion
- Spark distribution is defined by the combination of Spark, Hadoop and Scala versions and verified by the package checksum,
```

  see [Download Apache Spark](https://spark.apache.org/downloads.html) and the [archive repo](https://archive.apache.org/dist/spark/) for more information.
  - `spark_version`: The Spark version to install (`3.3.0`).
  - `hadoop_version`: The Hadoop version (`3.2`).
  - `scala_version`: The Scala version (`2.13`).
  - `spark_checksum`: The package checksum (`BFE4540...`).

> - Spark can run with different OpenJDK versions and those version need to match else Spark will not work.

Reviewer: This sentence is not clear to me. Could you take a look please?

Reviewer: Great, thank you! 👍

  - `openjdk_version`: The version of (JRE headless) the OpenJDK distribution (`17`), see [Ubuntu packages](https://packages.ubuntu.com/search?keywords=openjdk).

> - Starting Spark 3.2 the distribution file contains Scala version, hence building different version will only work for Spark >= 3.2
> - Building Spark < 3.2 requires modification to the Docker file or using an older version of the Dockerfile

Reviewer: ```suggestion
- Starting with _Spark >= 3.2_ the distribution file contains Scala version, hence building older Spark will not work.
- Building older version requires modification to the Dockerfile or using it's older version
```

Reviewer 2: Can we change _Spark >= 3.2_ to Spark >= 3.2?

Reviewer: I'm ok with both options :)

Reviewer 2: Oh.. its styling.. hehe I did't know :)

Reviewer: Using italic just makes it clear that `>= 3.2` is about spark version.
